oak_functions/examples/weather_lookup/module/src/tests.rs

```diff
@@ -158,3 +162,55 @@ fn test_location_from_slice() {
     assert_eq!(bytes, value.to_bytes(),);
     assert_eq!(value, Location::from_bytes(&bytes));
 }
+
+// This benchmark test takes a very long time, so is not run during tests.
+// To actually run the bench, use `cargo bench --features run-internal-bench`.
+#[cfg_attr(feature = "run-internal-bench", bench)]
+#[cfg_attr(not(feature = "run-internal-bench"), allow(dead_code))]
+fn bench_wasm_handler(bencher: &mut Bencher) {
+    let mut manifest_path = std::env::current_dir().unwrap();
+    manifest_path.push("Cargo.toml");
+    let wasm_module_bytes =
+        test_utils::compile_rust_wasm(manifest_path.to_str().expect("Invalid target dir"))
+            .expect("Couldn't read Wasm module");
+    let mut rng = rand::thread_rng();
+    let buf = generate_and_serialize_sparse_weather_entries(&mut rng, 10_000).unwrap();
+    let entries = parse_lookup_entries(buf).unwrap();
+    let lookup_data = Arc::new(LookupData::for_test(entries));
+
+    let summary = bencher.bench(|bencher| {
+        let logger = Logger::for_test();
+        let wasm_handler = WasmHandler::create(
+            &wasm_module_bytes,
+            lookup_data.clone(),
+            Arc::new(None),
+            logger,
+            None,
+        )
+        .expect("Couldn't create the server");
+        let rt = tokio::runtime::Runtime::new().unwrap();
```

Comment:
Is it possible to reuse tokio runtime? Does `Runtime` creation add up to the elapsed time?

Reply:
Ah, it probably should be counted to, since this is what we do upon user requests anyway.

Reply:
I actually think this should be outside, I remember it takes a nontrivial amount of time; and why do we create a new runtime per user anyways? I hope that's not actually the case, otherwise I suspect it may mess up the thread scheduling.

Reply:
Is not it already outside the loop? I think the bencher only repeats what is in `bencher.iter()`.

Reply:
Ah yes, my mistake, thanks for checking!

Reply:
I have now moved all setup code outside of the closure for clarity.